fearless_simd/src/macros.rs

```diff
@@ -140,50 +152,75 @@ macro_rules! simd_dispatch {
 /// #[inline(always)]
 /// fn sigmoid<S: Simd>(simd: S, x: &[f32], out: &mut [f32]) { /* ... */ }
 ///
-/// dispatch!(Level::new(), { #[inline(always)] |simd| sigmoid(simd, &[/*...*/], &mut [/*...*/]) });
+/// let level = Level::new();
+///
+/// dispatch!(level, simd => sigmoid(simd, &[/*...*/], &mut [/*...*/]));
 /// ```
 ///
 /// [`Level`]: crate::Level
 /// [`Level::new()`]: crate::Level::new
+/// [`Simd`]: crate::Simd
 #[macro_export]
 macro_rules! dispatch {
-    ($level:expr, $fn:expr) => {{
+    ($level:expr, $simd:pat => $op:expr) => {{
+        /// Convert the `Simd` value into an `impl Simd`, which enforces that
+        /// it is correctly handled.
         #[inline(always)]
         fn launder<S: $crate::Simd>(x: S) -> impl $crate::Simd {
             x
         }
 
         match $level {
-            $crate::Level::Fallback(fb) => ($fn)(launder(fb)),
+            $crate::Level::Fallback(fb) => {
+                let $simd = launder(fb);
+                // This vectorize call does nothing, but it is reasonable to be consistent here.
+                fb.vectorize(
```

Comment:
Maybe I am misunderstanding, but should the result of `launder` be used?
```suggestion
                let $simd = launder(fb);
                // This vectorize call does nothing, but it is reasonable to be consistent here.
                $simd.vectorize(
```